src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -703,6 +703,14 @@ private bool ShouldShareSymbol(ObjectNode node)
             if (!_targetPlatform.IsWindows)
                 return false;
 
+            // Types and methods from the compiler generated assembly are always shareable
+            CompilerGeneratedType type = (node is EETypeNode ? ((EETypeNode)node).Type : (node as MethodCodeNode)?.Method.OwningType) as CompilerGeneratedType;
```

Comment:
If you're checking for the `GeneratedAssembly`, we should probably cast to `MetadataType` here. Otherwise it's not much of an improvement.